website/content/en/highlights/2022-05-03-0-22-0-upgrade-guide.md

```diff
@@ -166,3 +170,430 @@ With this change, the message would be encoded as:
 ```text
 lvl=info msg="{\"some\":\"val\"}"
 ```
+
+### Deprecations
+
+#### Imminent removal of deprecated transforms replaced by `remap` and `reduce` {#deprecated-transforms}
+
+When the `remap` transform was introduced, several transforms were deprecated
+and removed from the documentation, despite still being available in Vector.
+Additionally the `merge` transform was replaced by `reduce`. In 0.23.0 we will
+be finally removing these transforms as part of some cleanup to our reference
+documentation.
+
+Transforms to be removed:
+
+- `add_fields`
+- `add_tags`
+- `ansi_stripper`
+- `coercer`
+- `concat`
+- `grok_parser`
+- `json_parser`
+- `key_value_parser`
+- `logfmt_parser`
+- `merge`
+- `regex_parser`
+- `remove_fields`
+- `remove_tags`
+- `rename_fields`
+- `split`
+- `tokenizer`
+
+See below for examples of how to replicate functionality using the `remap`
+transform.
+
+##### `add_fields`
+
+Before:
+
+```toml
+[transforms.add_fields]
+type = "add_fields"
+inputs = ["some_input"]
+fields.parent.child2 = "value2"
+```
+
+After:
+
+```toml
+[transforms.add_fields]
+type = "remap"
+inputs = ["some_input"]
+source = '''
+.parent.child2 = "value2"
+'''
+```
+
+##### `add_tags`
+
+Before:
+
+```toml
+[transforms.add_tags]
+type = "add_tags"
+inputs = ["some_input"]
+tags.some_tag = "some_value"
+```
+
+After:
+
+```toml
+[transforms.add_tags]
+type = "remap"
+inputs = ["some_input"]
+source = '''
+.tags = "some_value"
+'''
+```
+
+##### `ansi_stripper`
+
+Before:
+
+```toml
+[transforms.ansi_stripper]
+type = "ansi_stripper"
+inputs = ["some_input"]
+```
+
+After:
+
+```toml
+[transforms.ansi_stripper]
+type = "remap"
+inputs = ["some_input"]
+drop_on_err = false
+source = '''
+.message = strip_ansi_escape_codes!(.message)
```

Comment:
Does this need the `!` given that it's infallible? Or does it somehow extend to the argument itself? i.e. it acts more like `strip_ansi_escape_codes(string!(.message))`.

Reply:
That was my thought, yeah. Glad I was on the right track. :)

Reply:
This is an example of argument type fallibility. If VRL doesn't know the type of the argument, in this case `.message`, it makes the function call fallible which is why it is `strip_ansi_escape_codes!`. What @spencergilbert has would work too though, and is probably clearer so I'd be fine with changing it to that.

Reply:
I think in general we should avoid error handling on functions that are listed as "infallible"

Reply:
This is more a documentation problem, though. We explicitly went this route (of delegating argument type fallibility to function calls) because of the clutter it otherwise creates with all the type-enforcement function calls around each argument.

So, my preference would be to use what we promote, which is to use the function call with `!`, even though I can see how using `string!` makes things more explicit.

Reply:
We've definitely had users be confused that `infallible` functions are suggesting that the need error handling - we should definitely revisit this to make sure it's clear to users, especially new ones.

This probably also ties back into the feedback that VRL can be "intimidating" upfront.